### PR TITLE
Add support for setting the token through the CLI

### PR DESCRIPTION
**Pull request by Kurozume**

### CODE REVIEW

File: FredBoat/src/main/java/fredboat/FredBoat.java [hunk 1]
```diff
@@ -101,7 +101,10 @@
 
     private static DatabaseManager dbManager;
 
+    public static String BACKUP_TOKEN;
+
     public static void main(String[] args) throws LoginException, IllegalArgumentException, InterruptedException, IOException, UnirestException {
+        BACKUP_TOKEN = args[0];
```
Comment (Shredder121) [inline FredBoat/src/main/java/fredboat/FredBoat.java]: The problem is that anyone can see the command line arguments of your app. :(
I'd recommend environment variables. (`System.getEnv("fredboat.backupToken")` or something)

**Comment by Frederikam** [comment]
This looks like a bad idea from a security perspective. You'd be able to see the token from htop if you specified it this way.

File: FredBoat/src/main/java/fredboat/Config.java [hunk 2]
```diff
@@ -62,7 +62,7 @@
     public static int HIKARI_TIMEOUT_MILLISECONDS = 1000;
 
     private final DistributionEnum distribution;
-    private final String botToken;
+    private String botToken;
```
Comment (Shredder121) [inline FredBoat/src/main/java/fredboat/Config.java]: I would keep it final and extend the `else botToken = ""` into its own body for the BACKUP_TOKEN.

**Comment by napstr** [comment]
1.

> Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 0
	at fredboat.FredBoat.main(FredBoat.java:107)

2.

This clashes with the legacy scope system.

**Comment by Frederikam** [comment]
>This clashes with the legacy scope system.
Which we should kill

**Pull request closed without merging**